src/libYARP_os/src/yarp/os/impl/windows/WinConsole.c

```diff
@@ -0,0 +1,21 @@
+#include <stdbool.h> // C99+
+#include <windows.h>
+
+void yarp_logger_enable_vt_colors(void)
+{
+    static bool enabled = false;
+
+    if (!enabled)
+    {
+        DWORD handleMode = 0;
+        HANDLE hStdout = GetStdHandle(STD_OUTPUT_HANDLE);
+
+        if (hStdout != INVALID_HANDLE_VALUE && GetConsoleMode(hStdout, &handleMode))
+        {
+            handleMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;

```

Comment:
This file successfully compiles on Windows7+VS2005 when replacing `ENABLE_VIRTUAL_TERMINAL_PROCESSING` with `0x0004` ([docs](https://docs.microsoft.com/en-us/windows/console/setconsolemode)). Then, `SetConsoleMode` simply returns `0` on failure.

Reply:
Cool, in this case I think we can go on and merge this!